gradle build fails due to missing properties

Hi!

Commands like `./gradlew build` fail with exception

```
./gradlew build
Starting a Gradle Daemon (subsequent builds will be faster)

> Configure project :
Project : => 'org.pcollections' Java module

FAILURE: Build failed with an exception.

* Where:
Build file '/home/roby/development/redhat-managed-kafka/upstream/pcollections/build.gradle' line: 57

* What went wrong:
A problem occurred evaluating root project 'pcollections'.
> Could not get unknown property 'ossrhUsername' for Credentials [username: null] of type org.gradle.internal.credentials.DefaultPasswordCredentials_Decorated.
... snip
BUILD FAILED in 6s
```

I think these dynamic properties need to be in a gradle.properties file, I couldn't find a way to pass them as system props.

The context is we are rebuilding pcollections from source and are currently generating a gradle.properties file with dummy values.

It would be convenient if we didn't have to supply the file to run builds.

Maybe using an alternate syntax they could have default values? The sonatype publish would fail unless you have credentials set up on the host machine.

```groovy
providers.gradleProperty('ossrhUsername').getOrElse('dummy')
```

Or is there some other way to build without a gradle.properties file?

Thank you!

--- a/build.gradle
+++ b/build.gradle
@@ -54,8 +54,8 @@ publishing {
       url version.endsWith('SNAPSHOT') ? snapshotsRepoUrl : releasesRepoUrl
 
       credentials {
-        username ossrhUsername
-        password ossrhPassword
+        username providers.gradleProperty('ossrhUsername').getOrElse('dummy')
+        password providers.gradleProperty('ossrhPassword').getOrElse('dummy')
       }
     }
   }

